metricproviders/prometheus/prometheus.go

```diff
@@ -140,12 +145,51 @@ func NewPrometheusProvider(api v1.API, logCtx log.Entry) *Provider {
 
 // NewPrometheusAPI generates a prometheus API from the metric configuration
 func NewPrometheusAPI(metric v1alpha1.Metric) (v1.API, error) {
+	secretKeys := []string{Address}
+	envValuesByKey := lookupKeysInEnv(secretKeys)
+	log.Infof("envValuesByKey: %v, len: %v", envValuesByKey, len(envValuesByKey))
+	if len(metric.Provider.Prometheus.Address) != 0 {
+		if !IsUrl(metric.Provider.Prometheus.Address) {
+			return nil, errors.New("prometheus address is not is url format")
+		}
+	} else if envValuesByKey[Address] != "" {
+		if IsUrl(envValuesByKey[Address]) {
+			metric.Provider.Prometheus.Address = envValuesByKey[Address]
+		} else {
+			return nil, errors.New("prometheus address is not is url format")
+		}
+	} else {
+		return nil, errors.New("prometheus address is not configured")
+	}
 	client, err := api.NewClient(api.Config{
 		Address: metric.Provider.Prometheus.Address,
 	})
 	if err != nil {
+		log.Errorf("Error in getting prometheus client: %v", err)
 		return nil, err
 	}
-
 	return v1.NewAPI(client), nil
 }
+
+func IsUrl(str string) bool {
+	u, err := url.Parse(str)
+	if err != nil {
+		log.Errorf("Error in parsing url: %v", err)
+	}
+	log.Debugf("Parsed url: %v", u)
+	return err == nil && u.Scheme != "" && u.Host != ""
+}
+
+func lookupKeysInEnv(keys []string) map[string]string {
+	valuesByKey := make(map[string]string)
+	for i := range keys {
+		key := keys[i]
+		formattedKey := strings.ToUpper(strings.ReplaceAll(key, "-", "_"))
+		log.Debugf("formattedKey: %v", formattedKey)
+		if value, ok := os.LookupEnv(fmt.Sprintf("%s", formattedKey)); ok {
+			valuesByKey[key] = value
+			log.Debugf("PROMETHEUS_ADDRESS: %v", value)
+		}
+	}
+	return valuesByKey
+}
```

Comment:
I don't understand the purpose of this function. Why aren't we just simply looking up an env var by its well-known, documented env var name? Can we delete this?

Reply:
This function is useful when we want to extract multiple strings from env variables such as auth headers, api keys other than just the address.. this is also converting formats to "_" in case there was a misconfiguration with "-"  in env variable .. but I agree in this case just getting from env variable is fine.